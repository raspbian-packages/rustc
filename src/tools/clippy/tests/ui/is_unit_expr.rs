#![feature(plugin)]
#![plugin(clippy)]
#![warn(unit_expr)]
#[allow(unused_variables)]

fn main() {
    // lint should note removing the semicolon from "baz"
    let x = {
        "foo";
        "baz";
    };


    // lint should ignore false positive.
    let y = if true {
        "foo"
    } else {
        return;
    };

    // lint should note removing semicolon from "bar"
    let z = if true {
        "foo";
    } else {
        "bar";
    };


    let a1 = Some(5);

    // lint should ignore false positive
    let a2 = match a1 {
        Some(x) => x,
        _ => {
            return;
        },
    };

    // lint should note removing the semicolon after `x;`
    let a3 = match a1 {
        Some(x) => {
            x;
        },
        _ => {
            0;
        },
    };
}
