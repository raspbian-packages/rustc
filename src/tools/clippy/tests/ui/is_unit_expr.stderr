error: This expression evaluates to the Unit type ()
  --> $DIR/is_unit_expr.rs:8:13
   |
8  |       let x = {
   |  _____________^
9  | |         "foo";
10 | |         "baz";
11 | |     };
   | |_____^
   |
   = note: `-D unit-expr` implied by `-D warnings`
note: Consider removing the trailing semicolon
  --> $DIR/is_unit_expr.rs:10:9
   |
10 |         "baz";
   |         ^^^^^^

error: This expression evaluates to the Unit type ()
  --> $DIR/is_unit_expr.rs:22:13
   |
22 |       let z = if true {
   |  _____________^
23 | |         "foo";
24 | |     } else {
25 | |         "bar";
26 | |     };
   | |_____^
   |
note: Consider removing the trailing semicolon
  --> $DIR/is_unit_expr.rs:25:9
   |
25 |         "bar";
   |         ^^^^^^

error: This expression evaluates to the Unit type ()
  --> $DIR/is_unit_expr.rs:40:14
   |
40 |       let a3 = match a1 {
   |  ______________^
41 | |         Some(x) => {
42 | |             x;
43 | |         },
...  |
46 | |         },
47 | |     };
   | |_____^
   |
note: Consider removing the trailing semicolon
  --> $DIR/is_unit_expr.rs:42:13
   |
42 |             x;
   |             ^^

error: aborting due to 3 previous errors

